lib/triton-tags.js

```diff
@@ -0,0 +1,229 @@
+}
+
+
+/**
+ * Validate the given Triton tag key and value.
+ *
+ * This differs from `parseTritonTagStr` in that the `val` argument should
+ * already be of the correct type, and no value is returned.
+ *
+ * @param {String} key
+ * @param {String|Number|Boolean} val
+ * @returns {null|String} If valid, `null` is returned. Otherwise an error
+ *      message is returned.
+ */
+function validateTritonTag(key, val) {
+    assert.string(key, 'key');
+
+    // Check type.
+    var expectedType = typeFromKey[key];
+    var actualType = typeof (val);
+    if (expectedType === undefined) {
+        return format('Unrecognized special triton tag "%s"', key);
+    } else if (expectedType !== actualType) {
+        return format('Triton tag "%s" must be a %s: %j (%s)',
+            key, expectedType, val, actualType);
+    }
+
+    // Validate.
+    var validator = validatorFromKey[key];
+    assert.func(validator, 'validator for tag ' + key);
+    return validator(val);
+}
+
+
+/**
+ * Parse a Triton tag from the given string value.
+ *
+ * @param {String} key: The Triton tag key/name.
+ * @param {String} str: The tag value to parse.
+ * @param {Function} cb: `function (err, val)`
+ *      On success, `err` is null and `val` is the parsed and validated
+ *      tag value. On failure `err` is an Error instance.
+ */
+function parseTritonTagStr(key, str, cb) {
+    assert.string(key, 'key');
+    assert.string(str, 'str');
+    assert.func(cb, 'cb');
+
+    // Ensure it is a known triton tag.
+    var type = typeFromKey[key];
+    if (type === undefined) {
+        cb(new Error(format('Unrecognized special triton tag "%s"', key)));
+        return;
+    }
+
+    // Convert from string to value of appropriate type.
+    var val;
+    switch (type) {
+    case 'string':
+        val = str;
+        break;
+    case 'boolean':
+        if (str === 'true') {
+            val = true;
+        } else if (str === 'false') {
+            val = false;
+        } else {
+            cb(new Error(format(
+                'Triton tag "%s" value must be "true" or "false": %j',
+                key, str)));
+            return;
+        }
+        break;
+    case 'number':
+        val = Number(str);
+        if (str.length === 0  // Guard against `Number('') === 0` wat.
+            || isNaN(val))
+        {
+            cb(new Error(format('Triton tag "%s" value must be a number: %j',
+                key, str)));
+            return;
+        }
+        break;
+    default:
+        throw new Error('unexpected Triton tag type: ' + type);
+    }
+
+    // Validate.
+    var validator = validatorFromKey[key];
+    assert.func(validator, 'validator for tag ' + key);
+    var errmsg = validator(val);
+
+    if (errmsg) {
+        cb(new Error(errmsg));
+    } else {
+        cb(null, val);
+    }
+}
+
+
+module.exports = {
+    isTritonTag: isTritonTag,
+    validateTritonTag: validateTritonTag,
```

Comment:
It isn't yet. It is the intended function to be used by VMAPI later today when I break this
out to a separate node-triton-tags module and have docker and vmapi share that.
However, I'd like to get this in sdc-docker first.